### PR TITLE
Fix guest power reboot failure for open-vm-tools tests

### PR DESCRIPTION
**Pull request by nanzhg**
Guest power reboot command failed, because the machine doesn't boot into the OS. To fix the issue, this PR will enable the grub timeout for checking the VM power state.
http://10.67.129.96/tests/2085#step/esxi_open_vm_tools/76

1. Fix getting vm_id failure
2. Provide a option 'KEEP_GRUB_TIMEOUT' to enable grub timeout
3. Use open-vm-tools command to disable VM time sync

- Related ticket: 
   https://progress.opensuse.org/issues/126860
   https://progress.opensuse.org/issues/127334
- Verification run: 
   svirt backend:  http://10.67.129.96/tests/2130
   qemu backend:  https://openqa.suse.de/tests/10825316

### CODE REVIEW

**Comment by nanzhg** [comment]
@alice-suse @xguo @waynechen55 @Julie-CAO @tbaev @tonyyuan1 @RoyCai7 @varunkojha please help to review this PR. Thanks

**Review by alice-suse — COMMENTED** [review]
@nanzhg  The PR looks good. 
About validation jobs, 
- http://10.67.129.96/tests/2115#step/esxi_open_vm_tools/190, it fails. Is it expected?
- besides, the PR touches qemu backend jobs. Better to also add some validation for it?

**Comment by nanzhg** [comment]
> @nanzhg The PR looks good. About validation jobs,
> 
> * http://10.67.129.96/tests/2115#step/esxi_open_vm_tools/190, it fails. Is it expected?
> * besides, the PR touches qemu backend jobs. Better to also add some validation for it?

Thanks for the review!
This failure is not related to the PR, it's a separate issue and needs more investigations.
Will run more validations for qemu backend.

**Comment by RoyCai7** [comment]
LGTM

**Comment by alice-suse** [comment]
> > @nanzhg The PR looks good. About validation jobs,
> > 
> > * http://10.67.129.96/tests/2115#step/esxi_open_vm_tools/190, it fails. Is it expected?
> > * besides, the PR touches qemu backend jobs. Better to also add some validation for it?
> 
> Thanks for the review! This job is based on qemu backend, and the failure is not related to the code change. It's a separate issue and needs more investigations.

Is there any misunderstanding? According to http://10.67.129.96/tests/2115#settings, it is svirt backend. Isn't it?

**Comment by nanzhg** [comment]
> > > @nanzhg The PR looks good. About validation jobs,
> > > 
> > > * http://10.67.129.96/tests/2115#step/esxi_open_vm_tools/190, it fails. Is it expected?
> > > * besides, the PR touches qemu backend jobs. Better to also add some validation for it?
> > 
> > 
> > Thanks for the review! This job is based on qemu backend, and the failure is not related to the code change. It's a separate issue and needs more investigations.
> 
> Is there any misunderstanding? According to http://10.67.129.96/tests/2115#settings, it is svirt backend. Isn't it?

Sorry, I mean the 2nd link. :) 
For sle-micro: https://openqa.suse.de/tests/10825316

File: tests/virt_autotest/esxi_open_vm_tools.pm [hunk 1]
```diff
@@ -324,19 +318,6 @@ sub do_clock_sync_tests {
...
-        $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
-        check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
-
-        revert_vm_timesync_setting($vm_name);
-
-        $powerops = 'power.on';
-        $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
-        check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
-    }
 }
```
Comment (waynechen55) [inline tests/virt_autotest/esxi_open_vm_tools.pm:321]: Why disable time/clock sync/test completely ?

Comment (nanzhg) [reply tests/virt_autotest/esxi_open_vm_tools.pm:321]: This code change does not disable the test, just using vm tools cmd for instead.

File: tests/virt_autotest/esxi_open_vm_tools.pm [hunk 2]
```diff
@@ -144,9 +144,11 @@ sub login_vm_console {
     reset_consoles;
     console('svirt')->start_serial_grab;
     select_console('sut');
-    assert_screen('grub2', 90);
-    send_key 'ret';
-    assert_screen('linux-login', 90);
+    unless (get_var('KEEP_GRUB_TIMEOUT')) {
+        assert_screen('grub2', 200);
+        send_key 'ret';
+    }
```
[the remark this answers is not in the document]
Comment (nanzhg) [reply tests/virt_autotest/esxi_open_vm_tools.pm:150]: Thank you for the info! Will check and update.

[the remark this answers is not in the document]
Comment (nanzhg) [reply tests/virt_autotest/esxi_open_vm_tools.pm]: Removed this section as it doesn't take effect.

**Comment by alice-suse** [comment]
@nanzhg  Is this  PR ready on your side?

**Comment by nanzhg** [comment]
> @nanzhg Is this PR ready on your side?

Yes, it's ready for merge.

**Pull request merged**